backend/controller/pubsub/testdata/java/subscriber/src/main/java/xyz/block/ftl/java/test/subscriber/Subscriber.java

```diff
@@ -0,0 +1,45 @@
+package xyz.block.ftl.java.test.subscriber;
+
+import ftl.builtin.CatchRequest;
+import ftl.publisher.PubSubEvent;
+import ftl.publisher.TestTopicSubscription;
+import ftl.publisher.Topic2Subscription;
+import io.quarkus.logging.Log;
+import xyz.block.ftl.Retry;
+import xyz.block.ftl.Subscription;
+import xyz.block.ftl.TopicDefinition;
+import xyz.block.ftl.Verb;
+import xyz.block.ftl.VerbName;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class Subscriber {
+
+    private static final AtomicInteger catchCount = new AtomicInteger();
+    @TestTopicSubscription
+    void consume(PubSubEvent event) throws Exception {
+        Log.infof("Subscriber is consuming %s", event.getTime());
+    }
+
+    @Subscription(
+            topic = "topic2",
+            module = "publisher",
+            name = "doomedSubscription"
```

Comment:
This should be typesafe ideally, is this just temporarily using string constants?

Reply:
The one above is typesafe, using the generated `@TestTopicSubscription`. I didn't rename this one (as the generated code has no way of knowing I want my subscription to be named like this, and also to test out the non-generated approach).

If this use case is common I guess I could add a `@SubscriptionName` annotation you could combine to override the auto generated one.

Reply:
Aaah I see, this is kind of a low-level API.

Reply:
Yea, it's also a meta-annotation, so the generated one looks like:

```
/**
 * Subscription to the topic of type {@link PubSubEvent}
 */
@Retention(java.lang.annotation.RetentionPolicy.RUNTIME)
@Subscription(
    topic = "topic2",
    module = "publisher",
    name = "topic2Subscription"
)
public @interface Topic2Subscription {
}
```